core/src/main/java/com/linecorp/armeria/server/http/auth/HttpAuthService.java

```diff
@@ -0,0 +1,63 @@
+package com.linecorp.armeria.server.http.auth;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.common.http.DefaultHttpResponse;
+import com.linecorp.armeria.common.http.HttpHeaderNames;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.DecoratingServiceFunction;
+
+final class HttpAuthService {
+    private static final Pattern AUTHORIZATION_HEADER_PATTERN = Pattern.compile(
+            "(?<mechanism>\\S+) (?<authorization>\\S+)");
+    private static final Decoder BASE_64_DECODER = Base64.getDecoder();
+
+    private static final Function<String, Entry<String, String>> basicAuthTokenExtractor = (String s) -> {
+        Matcher matcher = AUTHORIZATION_HEADER_PATTERN.matcher(s);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("Invalid authorization header: " + s);
```

Comment:
How about logging at debug level and returning `null` instead? Otherwise a bad authrozation header will cause a 500 error.